src/modules/mim/firewall.json

```diff
@@ -1,41 +1,305 @@
+                      "name": "delete",
+                      "enumValue": 2
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "command",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "accept",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "reject",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "drop",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "type",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "inbound",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "outbound",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "forward",
+                      "enumValue": 3
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "protocol",
+                "schema": {
+                  "type": "enum",
+                  "valueSchema": "integer",
+                  "enumValues": [
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "all",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "tcp",
+                      "enumValue": 2
+                    },
+                    {
+                      "name": "udp",
+                      "enumValue": 3
+                    },
+                    {
+                      "name": "icmp",
+                      "enumValue": 4
+                    }
+                  ]
+                }
+              },
+              {
+                "name": "source",
+                "schema": "string"
+              },
+              {
+                "name": "sourcePort",
+                "schema": "integer"
+              },
+              {
+                "name": "destination",
+                "schema": "string"
+              },
+              {
+                "name": "destinationPort",
+                "schema": "integer"
+              },
```

Comment:
Why not just one "port", either destination or source depending on rule direction?

Reply:
This is because the port is not known just by the rule direction. For example, rules to allow all incoming SSH traffic will look like this:
```
sudo iptables -A INPUT -p tcp --dport 22 -m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT
sudo iptables -A OUTPUT -p tcp --sport 22 -m conntrack --ctstate ESTABLISHED -j ACCEPT
```
however, a rule to allow outgoing SSH traffic will look like this:
```
sudo iptables -A OUTPUT -p tcp --dport 22 -m conntrack --ctstate NEW,ESTABLISHED -j ACCEPT
sudo iptables -A INPUT -p tcp --sport 22 -m conntrack --ctstate ESTABLISHED -j ACCEPT
```

In the first example incoming traffic (`INPUT`) uses the destination port (`-dport`) while and outgoing traffic (`OUTPUT`) uses the source port (`--sport`). But in the second example it is reversed (outgoing -> destination port, incoming -> source port).

Reply:
This also shows the uses for `state` (`NEW`, `ESTABLISHED`)

Reply:
An out rule does not need source info - the source is the local device,
An in rule does not need destination info - the destination is the local device.
Simplify and merge the two sets of ports and targets into one, whose direction is dictated by the rule kind, in or out.

Reply:
We discussed to keep 2 sets